include/tester/vtcp/tcp/TCPServer.hpp

```diff
@@ -51,31 +40,37 @@ class TCPServer
 protected:    
     boost::asio::io_context _context;
 
-    std::unordered_map<std::size_t, std::shared_ptr<SessionType>> _sessions;
+    std::unordered_map<std::size_t, std::shared_ptr<TCPSession>> _sessions;
+
+    virtual void handle(const std::size_t id, const Message& request) = 0;
+
+    void send(const std::size_t id, const Message& message) 
+    {
+        try
+        {
+            _sessions.at(id)->send(message);
+        }
+        catch(const std::exception& e)
+        {
+            std::cerr << e.what() << std::endl;
+        }
+    }
 
 private:
     void onClientConnected(const std::size_t id, tcp::socket socket)
     {
         std::cout << "Accepted connection from " << socket.remote_endpoint().address().to_string()
             << " : " << socket.remote_endpoint().port() << std::endl;
-        auto session = std::make_shared<SessionType>(std::move(socket), id,
+        auto session = std::make_shared<TCPSession>(std::move(socket), id,
```

Comment:
Swap places between `id` and `std::move(socket)` (also in the constructor of `TCPSession`)

Reply:
Done